scripts/templates/generate_coverage_report

```diff
@@ -125,24 +129,38 @@ if [[ "$1" == "--codecov" ]]; then
       echo "WARNING: No CODECOV_TOKEN defined. Upload to CodeCov might fail"
    fi
 
-   # Remove other files to be sure CodeCov uses only the coverage.info file
-   rm -f lcov.info
-   remove_gcda_files
-
    # Upload report to CodeCov
+   if [ ! -d build/tmp ]; then
+      mkdir build
+      mkdir build/tmp
+   fi
 
-   if ! bash <(curl -s https://codecov.io/bash) -X gcov; then
+   cd build/tmp/
```

Comment:
**BUG:** Don't change the directory like that. It might not work.
Do `pushd $BUILD/tmp` and later, after the chmod, do `popd`, to the restore the CWD.
Also, later run codecov from the current directory with: `$BUILD/tmp/codecov`.